connect-inject/consul_sidecar.go

```diff
@@ -12,7 +12,7 @@ import (
 // It always disables service registration because for connect we no longer
 // need to keep services registered as this is handled in the endpoints-controller.
 func (h *Handler) consulSidecar(pod corev1.Pod) (corev1.Container, error) {
-	run, err := h.shouldRunMergedMetricsServer(pod)
+	run, err := h.InjectConfig.shouldRunMergedMetricsServer(pod)
```

Comment:
here is an idea: can we create a method on `InjectConfig` that returns all the ports you need if `run=true` but an `error` otherwise? We seem to make 4 distinct calls to `h.InjectConfig` which made me wonder if we could just do it in 1?

Reply:
Yeah good point. I had this same thought in the original code, would be nice to just have a function to return all the ports!